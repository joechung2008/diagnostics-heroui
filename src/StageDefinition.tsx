import {
  Table,
  TableBody,
  TableCell,
  TableColumn,
  TableHeader,
  TableRow,
} from "@heroui/react";

const StageDefinition: React.FC<StageDefinitionProps> = ({
  stageDefinition,
}) => {
  const items = Object.entries(stageDefinition).reduce<
    KeyValuePair<string[]>[]
  >((previous, [key, value]) => [...previous, { key, value }], []);

  return (
    <div>
      <h2 className="text-2xl font-semibold" id="stage-definitions">
        Stage Definitions
      </h2>
      <Table aria-labelledby="stage-definitions" role="table">
        <TableHeader>
          <TableColumn id="stage-key-header" scope="col">
            Key
          </TableColumn>
          <TableColumn id="stage-value-header" scope="col">
            Value
          </TableColumn>
        </TableHeader>
        <TableBody>
          {items.map((item) => (
            <TableRow key={item.key}>
              <TableCell role="cell" headers="stage-key-header">
                {item.key}
              </TableCell>
              <TableCell role="cell" headers="stage-value-header">
                {item.value.join(", ")}
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </div>
  );
};

export default StageDefinition;
