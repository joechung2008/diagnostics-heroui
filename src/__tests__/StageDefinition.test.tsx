import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import StageDefinition from "../StageDefinition";

describe("StageDefinition", () => {
  it("matches snapshot", () => {
    const mockStageDefinition = {
      development: ["dev1", "dev2"],
      production: ["prod1"],
      staging: ["stage1", "stage2", "stage3"],
    };
    const { asFragment } = render(
      <StageDefinition stageDefinition={mockStageDefinition} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
