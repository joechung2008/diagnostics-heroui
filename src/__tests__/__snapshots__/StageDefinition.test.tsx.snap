// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StageDefinition > matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="text-2xl font-semibold"
      id="stage-definitions"
    >
      Stage Definitions
    </h2>
    <div
      class="flex flex-col relative gap-4 w-full"
    >
      <div
        class="p-4 z-0 flex flex-col relative justify-between gap-4 bg-content1 overflow-auto shadow-small rounded-large w-full"
      >
        <table
          aria-describedby=""
          aria-labelledby="stage-definitions"
          class="min-w-full h-auto table-auto w-full"
          data-collection="react-aria-«r1»"
          id="react-aria-«r0»"
          role="table"
          tabindex="0"
        >
          <thead
            class="[&>tr]:first:rounded-lg"
            role="rowgroup"
          >
            <tr
              class="group/tr outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
              role="row"
            >
              <th
                aria-colindex="1"
                class="group/th px-3 h-10 align-middle bg-default-100 whitespace-nowrap text-foreground-500 text-tiny font-semibold first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start"
                data-collection="react-aria-«r1»"
                data-key="$.0"
                data-react-aria-pressable="true"
                id="stage-key-header"
                role="columnheader"
                scope="col"
                tabindex="-1"
              >
                Key
              </th>
              <th
                aria-colindex="2"
                class="group/th px-3 h-10 align-middle bg-default-100 whitespace-nowrap text-foreground-500 text-tiny font-semibold first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start"
                data-collection="react-aria-«r1»"
                data-key="$.1"
                data-react-aria-pressable="true"
                id="stage-value-header"
                role="columnheader"
                scope="col"
                tabindex="-1"
              >
                Value
              </th>
            </tr>
            <tr
              aria-hidden="true"
              class="w-px h-px block"
              style="margin-left: 0.25rem; margin-top: 0.25rem;"
              tabindex="-1"
            />
          </thead>
          <tbody
            class="after:block"
            role="rowgroup"
          >
            <tr
              aria-labelledby="react-aria-«r0»-development-$.0"
              class="group/tr outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
              data-collection="react-aria-«r1»"
              data-first="true"
              data-key="development"
              id="react-aria-«r5»"
              role="row"
              tabindex="-1"
            >
              <td
                class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:pointer-events-none before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]/tr:text-foreground-300 group-data-[disabled=true]/tr:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start"
                data-collection="react-aria-«r1»"
                data-key="development.0"
                headers="stage-key-header"
                id="react-aria-«r0»-development-$.0"
                role="cell"
                tabindex="-1"
              >
                <span>
                  development
                </span>
              </td>
              <td
                class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:pointer-events-none before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]/tr:text-foreground-300 group-data-[disabled=true]/tr:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start"
                data-collection="react-aria-«r1»"
                data-key="development.1"
                headers="stage-value-header"
                id="react-aria-«r7»"
                role="cell"
                tabindex="-1"
              >
                <span>
                  dev1, dev2
                </span>
              </td>
            </tr>
            <tr
              aria-labelledby="react-aria-«r0»-production-$.0"
              class="group/tr outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
              data-collection="react-aria-«r1»"
              data-key="production"
              data-middle="true"
              data-odd="true"
              id="react-aria-«r8»"
              role="row"
              tabindex="-1"
            >
              <td
                class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:pointer-events-none before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]/tr:text-foreground-300 group-data-[disabled=true]/tr:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start"
                data-collection="react-aria-«r1»"
                data-key="production.0"
                headers="stage-key-header"
                id="react-aria-«r0»-production-$.0"
                role="cell"
                tabindex="-1"
              >
                <span>
                  production
                </span>
              </td>
              <td
                class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:pointer-events-none before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]/tr:text-foreground-300 group-data-[disabled=true]/tr:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start"
                data-collection="react-aria-«r1»"
                data-key="production.1"
                headers="stage-value-header"
                id="react-aria-«ra»"
                role="cell"
                tabindex="-1"
              >
                <span>
                  prod1
                </span>
              </td>
            </tr>
            <tr
              aria-labelledby="react-aria-«r0»-staging-$.0"
              class="group/tr outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
              data-collection="react-aria-«r1»"
              data-key="staging"
              data-last="true"
              id="react-aria-«rb»"
              role="row"
              tabindex="-1"
            >
              <td
                class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:pointer-events-none before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]/tr:text-foreground-300 group-data-[disabled=true]/tr:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start"
                data-collection="react-aria-«r1»"
                data-key="staging.0"
                headers="stage-key-header"
                id="react-aria-«r0»-staging-$.0"
                role="cell"
                tabindex="-1"
              >
                <span>
                  staging
                </span>
              </td>
              <td
                class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:pointer-events-none before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]/tr:text-foreground-300 group-data-[disabled=true]/tr:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start"
                data-collection="react-aria-«r1»"
                data-key="staging.1"
                headers="stage-value-header"
                id="react-aria-«rd»"
                role="cell"
                tabindex="-1"
              >
                <span>
                  stage1, stage2, stage3
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
